ci2/ci/github.py

```diff
@@ -90,32 +106,53 @@ def to_dict(self):
         return {'repo': self.repo.to_dict(), 'name': self.name}
 
 
-class PR(object):
-    def __init__(self, number, title, source_sha, target_branch):
+class PR:
+    def __init__(self, number, title, source_repo, source_sha, target_branch):
         self.number = number
         self.title = title
+        self.source_repo = source_repo
         self.source_sha = source_sha
         self.target_branch = target_branch
+        self.sha = None
 
         # one of pending, changes_requested, approve
         self.review_state = None
 
         self.batch = None
-        self.passing = None
+        self.build_state = None
```

Comment:
build_state batch and sha must updated in sync, so i prefer self.sha setting down here much as it is elsewhere that we set all three to none